File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/bookFilterRequest.ts
import { CoverType, LibraryFilterRequest } from "../../..";

export enum BookSorting {
    MostPopular, LeastPopular
}

export interface BookFilterRequest extends LibraryFilterRequest {
    publicationFrom: Date | null;
    publicationTo: Date | null;
    minPrice: number | null;
    maxPrice: number | null;
    coverType: CoverType | null;
    onlyInStock: boolean | null;
    minPageAmount: number | null;
    maxPageAmount: number | null;
    authorId: number | null;
    genreId: number | null;
    publisherId: number | null;
    sorting: BookSorting | null;
}

export function defaultBookFilterRequest(): BookFilterRequest {
    return {
        containsName: "",
        pageNumber: 0,
        pageSize: 0,
        publicationFrom: null,
        publicationTo: null,
        minPrice: null,
        maxPrice: null,
        coverType: null,
        onlyInStock: false,
        minPageAmount: null,
        maxPageAmount: null,
        authorId: null,
        genreId: null,
        publisherId: null,
        sorting: null,
    };
}

File: src/ELibrary.Backend/LibraryShopEntities/Filters/BookFilterRequest.cs
using LibraryShopEntities.Domain.Entities.Library;

namespace LibraryShopEntities.Filters
{
    public enum BookSorting
    {
        MostPopular, LeastPopular
    }

    public class BookFilterRequest : LibraryFilterRequest
    {
        public DateTime? PublicationFrom { get; set; }
        public DateTime? PublicationTo { get; set; }
        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }
        public CoverType? CoverType { get; set; }
        public bool? OnlyInStock { get; set; }
        public int? MinPageAmount { get; set; }
        public int? MaxPageAmount { get; set; }
        public int? AuthorId { get; set; }
        public int? GenreId { get; set; }
        public int? PublisherId { get; set; }
        public BookSorting? Sorting { get; set; }

        public override string ToString()
        {
            return $"PageNumber: {PageNumber}, PageSize: {PageSize}, ContainsName: '{ContainsName}', " +
                   $"PublicationFrom: {PublicationFrom?.ToString("yyyy-MM-dd") ?? "null"}, " +
                   $"PublicationTo: {PublicationTo?.ToString("yyyy-MM-dd") ?? "null"}, " +
                   $"MinPrice: {MinPrice?.ToString("F2") ?? "null"}, MaxPrice: {MaxPrice?.ToString("F2") ?? "null"}, " +
                   $"CoverType: {CoverType?.ToString() ?? "null"}, OnlyInStock: {OnlyInStock?.ToString() ?? "null"}, " +
                   $"MinPageAmount: {MinPageAmount?.ToString() ?? "null"}, MaxPageAmount: {MaxPageAmount?.ToString() ?? "null"}, " +
                   $"AuthorId: {AuthorId?.ToString() ?? "null"}, GenreId: {GenreId?.ToString() ?? "null"}, " +
                   $"PublisherId: {PublisherId?.ToString() ?? "null"}, Sorting: {Sorting?.ToString() ?? "null"}";
        }
    }
}